modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheIoManager.java

```diff
@@ -1206,10 +1206,10 @@ public void send(ClusterNode node, GridCacheMessage msg, byte plc) throws Ignite
 
         while (cnt <= retryCnt) {
             try {
-                cnt++;
-
                 cctx.gridIO().sendToGridTopic(node, TOPIC_CACHE, msg, plc);
 
+                cnt++;
```

Comment:
Do we actually need to increment counter here? Let's increment it on retries only.

Reply:
Good point! Thank you.